internal/kubeclient/indexer.go

```diff
@@ -36,6 +39,25 @@ const (
 	ServiceAccountsByOIDCSubjectIndexField = "subjects"
 )
 
+func IndexEventsByInvolvedObjectAPIGroup(ctx context.Context, mgr ctrl.Manager) error {
+	return mgr.GetFieldIndexer().IndexField(
+		ctx,
+		&corev1.Event{},
+		EventsByInvolvedObjectAPIGroupIndexField,
+		indexEventsByInvolvedObjectAPIGroup,
+	)
+}
+
+func indexEventsByInvolvedObjectAPIGroup(obj client.Object) []string {
+	event := obj.(*corev1.Event) // nolint: forcetypeassert
+	// Ignore invalid APIVersion
+	gv, _ := schema.ParseGroupVersion(event.InvolvedObject.APIVersion)
+	if gv.Empty() {
+		return nil
+	}
+	return []string{gv.Group}
+}
```

Comment:
Another idea for here is to create `kargoEvent` field with (string formatted) boolean value to distinguish Kargo related events or not. It might be helpful when we introduce other API group other than `kargo.akuity.io` 🤔

Reply:
That's a good idea, but I think in the interest of getting this into 0.6, we should make this change as a later improvement